### ISSUE TITLE
Reproduction-related queries

### ISSUE THREAD

**Issue by Ysiz** [suchang-99/mmWaveFlow#1]
# Issue: Encountered Several Issues During Reproduction

Dear @suchang-99 ,

Thank you for open\-sourcing this excellent project\! I encountered some problems while reproducing the paper's experiments and hope to get your help\.

The third issue below, **`3. (Urgent!) Parameter Configuration Inconsistent with the Paper`**, I hope you can reply to it with priority\.

## 1\. Environment Configuration

- Description:

    I strictly followed the `requirements.txt` and first installed **PyTorch 1\.12\.1 \+ CUDA 11\.3**, but then encountered issues when installing other dependencies:

    1. `einops == 0.8.2` and `scipy == 1.17.1` are incompatible with PyTorch 1\.12, so I downgraded them to lower versions: `einops==0.8.1` and `scipy==1.10.1`\.

    2. An error occurred during the compilation of PyTorchEMD, which also seems to be related to the PyTorch version\. I modified some header files in PyTorchEMD to solve this problem, which should not affect the EMD calculation\.

- **Request:**

    I noticed that some dependency versions in `requirements.txt` are incompatible with `PyTorch 1.12.1`\. Could you confirm the correct environment configuration?

## 2\. Datasets Files Missing and Preprocessing Issues

I encountered the following problems when training with mRI and mmFi datasets:

#### Problem 1: Missing Preprocessing Files

- Description:

    The repository does not provide the necessary preprocessing files:

    - `J_regression_coco.npy`
    -  **GMM prior file**

     I downloaded files with the same names from open\-source SMPL\-related repositories, but cannot confirm that the file parameters and mean weights are exactly consistent with those used in the original paper\.

- **Request:**

    I hope the author can provide these two standard files; if convenient, you can directly provide the preprocessed `all_data_list.npy` to avoid errors introduced by the preprocessing process\.

#### Problem 2: SMPL Model Version

- Description:

    The original paper uses `basicModel_neutral_lbs_10_207_0_v1.0.0.pkl`, which i failed to find on SMPL website\. I obtained this file from another open\-source project, and there is a risk of version mismatch\.

- **Request:**

    I hope you can supplement the `basicModel_neutral_lbs_10_207_0_v1.0.0.pkl` file or it’s official download link\.

### 2\. mmFi Preprocessing\-Related Issues

- Description:

    I encountered problems during the data preprocessing stage, but did not keep the complete error log\. However, I modified `MMFi_DataLoader.py` and `universal_utils.py` to make the preprocessing run\. **I will submit the specific content of my modifications in another Issue**\.

- **Request:**

    - If convenient, please check whether the logic of my modifications complies with your preprocessing standards\.

    - Could you please provide the following preprocessed files if convenient: `test_list.npy`, `test_mmwave_num.npy`, `train_list.npy`, `train_mmwave_num.npy`, to verify whether the output of my local preprocessing is correct\.

## 3\. (Urgent\!)Parameter Configuration Inconsistent with the Paper

**After sloving problems mentioned above, I was able to run the training on the mRI and mmFi datasets\. However, after running 200 epochs on mRI and 100 epochs on mmFi, I found that the loss did not converge\. I then checked the parameter configuration in the `yaml` files and found the following questions:**

### Problem 1

- Description:

    In **5\.1\.3\. Implementation Details** of your paper, it is mentioned that:

    > The latent comprises N = 32 tokens and the shared query set consists of M = 16 tokens\. All tokens have dimensionality D = 128…… $\lambda_{\text{KL}} = 1 \times 10^{-5}$ and $\lambda _{\text{VAE}}$ , $\lambda _{\text{Flow}}$ and $\lambda _{\text{align}}$ in Eq\. \(7\) are 50, 1 and 0\.1, respectively\.
    > 
    
    However, I found that the code you provided seems to have some differences from the paper, including the following points:
    
    - `config.yaml` **lines 8\-9**: `latent_num: 8 latent_dim: 64` – Are these two parameters written correctly?
    
    - `mri.yaml` **line 14**: `z_flow:0.1` – this seems inconsistent with $\lambda _{\text{Flow}}=1$ mentioned in the paper?
    
    - `mWGModel.py` **line 149**:
    
        ```python
        self.cross_attention=CrossAttnEncoder(d_model=latent_dim,nhead=8,n_query=latent_num,n_layers=1)
        ```
    
        You set `n_query=latent_num`, which is inconsistent with **the shared query set consists of M = 16 tokens** in the paper?
    
        **Summary in the table below:**
    
        |        Paper Description|        Code Configuration|        Location|
        |---|---|---|
        |        latent\_num = 32|        latent\_num = 8|        config\.yaml line 8|
        |        latent\_dim = 128|        latent\_dim = 64|        config\.yaml line 9|
        |        λ\_Flow = 1|        z\_flow = 0\.1|        mri\.yaml line 14|
        |        M = 16 tokens|        n\_query = latent\_num| mWGModel\.py line 149 |

### Problem 2

In addition to the parameter settings explicitly mentioned in the paper, I have the following questions:

- Description:

    - Is `num_workers = 0` in `config.yaml line 25,29` a correct configuration for formal training?

    - Except for the above marked parameters, are all **other weights, optimizer parameters, etc**\. in the `yaml` files fully consistent with the paper?

    - When training on different datasets, **do the configurations in `config.yaml` need to be modified specifically on different datastes?**

### My Operations:

I have manually modified the above parameters and retrained, but the loss still shows no tendency to converge\.

### **Request:**

1. Please confirm whether the parameter settings in the code contain errors, or if the paper simplified the parameter descriptions, or if there exists an internal compensation mechanism for parameters within the program.

2. Please check if other training hyperparameters in yaml files and hard\-coded model parameters are correct

## 4\. CD Metric Value Magnitude Differs Greatly from the Paper

- Description:

    During training, I found that the **CD values** in **Table 1\.** of the paper differ by nearly 100 times from the **CD values output by the program**(e.g. The mRI M2D CD values I got are like 0.009，while the value in the table is 0.72)

- **Request:**

    Is there any difference between the **CD calculation method** in the program and the indicator in **Table 1\.** of your paper, such as a difference in units or a certain conversion relationship?

---

I have checked the entire process to the best of my ability but still cannot reproduce the results close to your paper\. 

**If there are any hidden configurations, training techniques, or preprocessing details in the project that are not marked in the documentation and need to be adjusted , please let me know\!** 

I apologize if there are any unreasonable aspects in the descriptions above.

Looking forward to your reply, best regards\.
